restapi/stories.go

```diff
@@ -72,6 +83,115 @@ func (api api) CreateStories(w http.ResponseWriter, r *http.Request) render.Rend
 	return rest.MsgStatusOK("Stories added successfully")
 }
 
+func (api api) s3upload(file multipart.File, size int64, name string) (string, error) {
+	newSession, err := session.NewSession(api.s3config)
+	if err != nil {
+		return "", fmt.Errorf("Could not connect with S3")
+	}
+
+	buffer := make([]byte, size) // read file content to buffer
+
+	_, err = file.Read(buffer)
+	if err != nil {
+		return "", fmt.Errorf("Could not read file")
+	}
+	s3Client := s3.New(newSession)
+	fileBytes := bytes.NewReader(buffer)
+	fileType := http.DetectContentType(buffer)
+	bucket := aws.String(os.Getenv("BUCKET_NAME"))
+
+	_, err = s3Client.PutObject(&s3.PutObjectInput{
+		Body:          fileBytes,
+		Bucket:        bucket,
+		Key:           aws.String(name),
+		ContentLength: aws.Int64(size),
+		ContentType:   aws.String(fileType),
+	})
+	if err != nil {
+		return "", fmt.Errorf("Could not upload to S3")
+	}
+	return fmt.Sprintf("https://%s.s3.us-west-000.backblazeb2.com/%s", os.Getenv("BUCKET_NAME"), name), nil
+}
+
+func convertYoutubeURL(originalURL string) (string, error) {
+	re := regexp.MustCompile(youtubeRegex)
+	match := re.FindAllStringSubmatch(originalURL, 2)
+	if len(match) == 0 {
+		return "", fmt.Errorf("Invalid Youtube Link")
+	}
+	return youtubeEmbedURL + match[0][1], nil
+}
+
+func (api api) CreateStoriesFormData(w http.ResponseWriter, r *http.Request) render.Renderer {
```

Comment:
hmm, what happens if we get an array of stories in this endpoint? Shouldn't we be looping through the array?